kate/recovery/src/com.rs

```diff
@@ -36,46 +37,40 @@ pub fn unflatten_padded_data(
 	layout: Vec<(u32, u32)>,
 	data: Vec<u8>,
 	chunk_size: usize,
-) -> Vec<(u32, Vec<u8>)> {
-	assert!(data.len() % chunk_size == 0);
-	let data = data
+) -> Result<Vec<(u32, Vec<u8>)>> {
```

Comment:
That could be an issue because `unfatten_padded_data` is used in `System::finalize` which should be **infallible**.